mm2src/adex_cli/src/adex_config.rs

```diff
@@ -0,0 +1,178 @@
+use anyhow::{anyhow, bail, Result};
+use directories::ProjectDirs;
+use inquire::Password;
+use log::{error, info, warn};
+use serde::{Deserialize, Serialize};
+use std::fmt::{Display, Formatter};
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use crate::helpers::rewrite_json_file;
+use crate::logging::{error_anyhow, warn_bail};
+
+const PROJECT_QUALIFIER: &str = "com";
+const PROJECT_COMPANY: &str = "komodoplatform";
+const PROJECT_APP: &str = "adex-cli";
+const ADEX_CFG: &str = "adex_cfg.json";
+
+const PRICE_PRECISION_MIN: usize = 8;
+const PRICE_PRECISION_MAX: usize = 8;
+const VOLUME_PRECISION_MIN: usize = 2;
+const VOLUME_PRECISION_MAX: usize = 5;
+const VOLUME_PRECISION: VolumePrecision = (VOLUME_PRECISION_MIN, VOLUME_PRECISION_MAX);
+const PRICE_PRECISION: PricePrecision = (PRICE_PRECISION_MIN, PRICE_PRECISION_MAX);
+
+pub(crate) type PricePrecision = (usize, usize);
+pub(crate) type VolumePrecision = (usize, usize);
+
+pub(crate) fn get_config() {
+    let Ok(adex_cfg) = AdexConfigImpl::from_config_path() else { return; };
+    info!("{}", adex_cfg)
+}
+
+pub(crate) fn set_config(set_password: bool, rpc_api_uri: Option<String>) {
+    let mut adex_cfg = AdexConfigImpl::from_config_path().unwrap_or_else(|_| AdexConfigImpl::default());
+    let mut is_changes_happened = false;
+    if set_password {
+        let rpc_password = Password::new("Enter RPC API password:")
+            .prompt()
+            .map(|value| {
+                is_changes_happened = true;
+                value
+            })
+            .map_err(|error| error_anyhow!("Failed to get rpc_api_password: {error}"))
+            .ok();
+        adex_cfg.set_rpc_password(rpc_password);
+    }
+    if rpc_api_uri.is_some() {
+        adex_cfg.set_rpc_uri(rpc_api_uri);
+        is_changes_happened = true;
+    }
+
+    if is_changes_happened && adex_cfg.write_to_config_path().is_ok() {
```

Comment:
What I understand is, if `set_password` is true, no matter what `rpc_password` will be updated.

And only if `rpc_api_uri.is_some()` true, `rpc_uri` will be updated.

If I am correct with above, then this should do the job:
```rs
    let mut adex_cfg = AdexConfigImpl::from_config_path().unwrap_or_else(|_| AdexConfigImpl::default());

    if set_password {
        let rpc_password = Password::new("Enter RPC API password:")
            .prompt()
            .map_err(|error| error_anyhow!("Failed to get rpc_api_password: {error}"))
            .ok();
        adex_cfg.set_rpc_password(rpc_password.clone());
    }

    let set_api_uri = rpc_api_uri.is_some();
    if set_api_uri {
        adex_cfg.set_rpc_uri(rpc_api_uri);
    }

    if set_password && set_api_uri && adex_cfg.write_to_config_path().is_ok() {
        info!("Configuration has been set");
    } else {
        warn!("Nothing changed");
    }
```

Reply:
[done](https://github.com/KomodoPlatform/atomicDEX-API/pull/1768/commits/fe0d4fb8084bbf8acfda5154578c75f22851d64e)